src/coreclr/jit/llvmcodegen.cpp

```diff
@@ -0,0 +1,1714 @@
+        case GT_CNS_LNG:
+            buildCnsLng(node);
+            break;
+        case GT_FIELD_LIST:
+            // does not build in linear order, but when the GT_FIELD_LIST is consumed
+            break;
+        case GT_IL_OFFSET:
+            _currentOffset = node->AsILOffset()->gtStmtILoffsx;
+            _currentOffsetDiLocation = nullptr;
+            break;
+        case GT_IND:
+            buildInd(node, getGenTreeValue(node->AsOp()->gtOp1));
+            break;
+        case GT_JTRUE:
+            buildJTrue(node, getGenTreeValue(node->AsOp()->gtOp1));
+            break;
+        case GT_LCL_FLD:
+            buildLocalField(node->AsLclFld());
+            break;
+        case GT_LCL_VAR:
+            localVar(node->AsLclVar());
+            break;
+        case GT_LCL_VAR_ADDR:
+        case GT_LCL_FLD_ADDR:
+            buildLocalVarAddr(node->AsLclVarCommon());
+            break;
+        case GT_LSH:
+        case GT_RSH:
+        case GT_RSZ:
+            buildShift(node->AsOp());
+            break;
+        case GT_EQ:
+        case GT_NE:
+        case GT_LE:
+        case GT_LT:
+        case GT_GE:
+        case GT_GT:
+            buildCmp(node, getGenTreeValue(node->AsOp()->gtOp1), getGenTreeValue(node->AsOp()->gtOp2));
+            break;
+        case GT_NEG:
+        case GT_NOT:
+            buildUnaryOperation(node);
+            break;
+        case GT_NO_OP:
+            emitDoNothingCall();
+            break;
+        case GT_NULLCHECK:
+            buildNullCheck(node->AsUnOp());
+            break;
+        case GT_OBJ:
+            buildObj(node->AsObj());
+            break;
+        case GT_PHI:
+            buildEmptyPhi(node->AsPhi());
+            break;
+        case GT_PHI_ARG:
+        case GT_PUTARG_TYPE:
+            break;
+        case GT_RETURN:
+            buildReturn(node);
+            break;
+        case GT_STORE_LCL_VAR:
+            storeLocalVar(node->AsLclVar());
+            break;
+        case GT_STOREIND:
+            buildStoreInd(node->AsStoreInd());
+            break;
+        case GT_STORE_BLK:
+        case GT_STORE_OBJ:
+            buildStoreBlk(node->AsBlk());
+            break;
+        case GT_AND:
+        case GT_OR:
+        case GT_XOR:
+            buildBinaryOperation(node);
+            break;
+        default:
+            failFunctionCompilation();
+    }
+}
+
+Value* Llvm::localVar(GenTreeLclVar* lclVar)
+{
+    Value*       llvmRef;
+    unsigned int lclNum = lclVar->GetLclNum();
+    unsigned int ssaNum = lclVar->GetSsaNum();
+    LclVarDsc*   varDsc = _compiler->lvaGetDesc(lclVar);
+
+    if (isLlvmFrameLocal(varDsc))
+    {
+        llvmRef = _builder.CreateLoad(m_allocas[lclNum]);
+    }
+    else if (_localsMap->find({lclNum, ssaNum}) == _localsMap->end())
+    {
+        if (varDsc->lvLlvmArgNum != BAD_LLVM_ARG_NUM)
+        {
+            // TODO-LLVM: we dont handle this yet
+            if (_info.compRetBuffArg != BAD_VAR_NUM)
```

Comment:
By the way, all of the handling of return buffers can just be deleted, since we have a purely "by-value" (to the frontend) ABI now.